src/psyclone/configuration.py

```diff
@@ -0,0 +1,426 @@
+        raise ConfigurationError("{0} not found in any of {1}".
+                                 format(_FILE_NAME, _file_paths))
+
+    @property
+    def distributed_memory(self):
+        '''
+        Getter for whether or not distributed memory is enabled
+        :returns: True if DM is enabled, False otherwise
+        :rtype: bool
+        '''
+        return self._distributed_mem
+
+    @distributed_memory.setter
+    def distributed_memory(self, dist_mem):
+        '''
+        Setter for whether or not distributed memory support is enabled
+        in this configuration.
+        :param bool dist_mem: Whether or not dm is enabled
+        '''
+        if not isinstance(dist_mem, bool):
+            raise ConfigurationError(
+                "distributed_memory must be a boolean but got {0}".
+                format(type(dist_mem)))
+        self._distributed_mem = dist_mem
+
+    @property
+    def default_api(self):
+        '''
+        Getter for the default API used by PSyclone.
+        :returns: default PSyclone API
+        :rtype: str
+        '''
+        return self._default_api
+
+    @property
+    def supported_apis(self):
+        '''
+        Getter for the list of APIs supported by PSyclone.
+        :returns: list of supported APIs
+        :rtype: list of str
+        '''
+        return self._supported_api_list
+
+    @property
+    def default_stub_api(self):
+        '''
+        Getter for the default API used by the stub generator.
+        :returns: default API for the stub generator
+        :rtype: str
+        '''
+        return self._default_stub_api
+
+    @property
+    def supported_stub_apis(self):
+        '''
+        Getter for the list of APIs supported by the stub generator.
+        :returns: list of supported APIs.
+        :rtype: list of str
+        '''
+        return self._supported_stub_api_list
+
+    @property
+    def reproducible_reductions(self):
+        '''
+        Getter for whether reproducible reductions are enabled.
+        :returns: True if reproducible reductions are enabled, False otherwise.
+        :rtype: bool
+        '''
+        return self._reproducible_reductions
+
+    @property
+    def reprod_pad_size(self):
+        '''
+        Getter for the amount of padding to use for the array required
+        for reproducible OpenMP reductions
+        :returns: padding size (no. of array elements)
+        :rtype: int
+        '''
+        return self._reprod_pad_size
+
+    @property
+    def filename(self):
+        '''
+        Getter for the full path and name of the configuration file used
+        to initialise this configuration object.
+        :returns: full path and name of configuration file
+        :rtype: str
+        '''
+        return self._config_file
+
+
+class DynConfig(object):
+    '''
+    Dynamo0.3-specific Config sub-class. Holds configuration options specific
+    to the Dynamo 0.3 API.
+
+    :param config: The 'parent' Config object.
```

Comment:
I note, you have started using the "describe __init__ args in the class description". I thought you didn't like this - and in fact this is not the way you have done it in the past? I think we can ignore for the moment as we don't seem to have one agreed style.